website/docs/document-api.md

```diff
@@ -146,6 +146,287 @@ markdown.equals(html);
 
 ## `insertText`
 
+### Usage
+
+```ts
+let doc = new Document({ content: "foo", annotations: [] });
+
+// insert text at position 3, with content of bar
+doc.insertText(3, "bar");
+
+// doc
+{ content: "foobar", annotations: []}
+```
+
+### With annotations
+
+The `insertText` method will adjust existing annotations when text is inserted
+into a document. For example, if you had this document with a bold annotation:
+
+```ts
+let doc = new Document({ content: "Foo bar" });
+doc.addAnnotations(new Bold({
+  start: 0
+  end: 7
+}));
+```
+
+If you were to call `insertText` to insert a string in the middle of the
+existing bold annotation, the annotation will be automatically extended to
+include the newly-inserted text and the existing surrounding text.
+
+```ts
+doc.insertText(4, "baz ");
+```
+
+Would result in this document:
+
+```ts
+{
+  content: "Foo baz bar"
+  annotations: [
+    {
+      type: "-example-bold",
+      start: 0,
+      end: 11,
+    }
+  ]
+}
+```
+
+### Adjacent boundary behaviour
+
+When text is inserted at either the leading or trailing edge of an annotation,
+atjson behaves a bit differently, providing a number of options to control how
+the annotation reacts to the inserted text. This adjacent boundary behaviour is
+controlled via an optional third argument to `insertText`. Valid values are:
+
+```ts
+import { AdjacentBoundaryBehaviour } from "@atjson/document";
+
+// assume the document from above
+
+// these three are functionally equivalent
+doc.insertText(4, "baz ");
+doc.insertText(4, "baz ", AdjacentBoundaryBehaviour.preserveLeading);
+// "modify" is targeted for deprecation
+doc.insertText(4, "baz ", AdjacentBoundaryBehaviour.modify);
+
+// these two are functionally equivalent
+doc.insertText(4, "baz ", AdjacentBoundaryBehaviour.preserveTrailing);
+// "preserve" is targeted for deprecation
+doc.insertText(4, "baz ", AdjacentBoundaryBehaviour.preserve);
+
+doc.insertText(4, "baz ", AdjacentBoundaryBehaviour.preserveBoth);
+```
+
+Each option is described below in detail.
+
+#### `preserveLeading` / `modify` (default)
+
+This is the default adjacent boundary behaviour. With the `preserveLeading`
+option, text that is inserted at a leading boundary of an annotation will not modify
+the annotation that the new text is adjacent to.
+
+Example:
+
+```ts
+let doc = new Document({ content: "Foo bar" });
+doc.addAnnotations(new Bold({
+  start: 4
+  end: 7
+}));
+```
+
+At this point, the document conceptually looks like this: (* represents the
```

Comment:
Does this makes sense / help? I wanted some way to show visually without having to open up photoshop

Reply:
Yes! This is great; it might be helpful to have an illustration, but this is at least complete and clear.

Reply:
Yes, we can also use some of the components that I wrote for this— I've done this in the "Getting Started" page.

Reply:
Nice! I've swapped those in where appropriate. I came across a bug when an annotation ends at the end of the text string and some margin expectations that make these look taller since there's no image here. I don't want to scope creep this PR so will leave those tweaks / fixes for another time. For the time being, I've just added a space at the end of my strings for docs I'm passing into the QuickBrownFox component

Reply:
😬 oh, lol

Reply:
Thank you for gracefully dealing with that wild code

Reply:
The docs look really nice!

We can improve them later, but I'd love to have the result be triggerable by a user (if that makes sense). A mini playground of sorts is what I was going for. This is a great first step :)